### PR TITLE
Repaired Metrics name collision between Privacy and RocksDB

### PR DESCRIPTION
**Pull request by rain-on**
When Privacy is enabled in Besu, a second database is created, however it duplicates the metrics used in the public state database - which results in a "collector" name collision - which in turn halts Besu operation.

The solution has been to duplicate the Metrics categories associated with the RocksDB, but prefixed with "privacy" - removing the name-collision.

### CODE REVIEW

File: metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java [hunk 1]
```diff
@@ -40,9 +42,11 @@
   public static final Set<MetricCategory> DEFAULT_METRIC_CATEGORIES;
 
   static {
-    // Why not ROCKSDB and KVSTORE_ROCKSDB_STATS?  They hurt performance under load.
+    // Why not ROCKSDB and KVSTORE_ROCKSDB_STATS, KVSTORE_PRIVATE_ROCKSDB_STATS?  They hurt
+    // performance under load.
     final EnumSet<BesuMetricCategory> besuCategories =
-        EnumSet.complementOf(EnumSet.of(KVSTORE_ROCKSDB, KVSTORE_ROCKSDB_STATS));
+        EnumSet.complementOf(
```
Comment (CjHare) [inline metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java]: Why is `KVSTORE_PRIVATE_ROCKSDB` not needed in the set but `KVSTORE_ROCKSDB` is?

File: metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java [hunk 2]
```diff
@@ -40,9 +42,11 @@
   public static final Set<MetricCategory> DEFAULT_METRIC_CATEGORIES;
 
   static {
-    // Why not ROCKSDB and KVSTORE_ROCKSDB_STATS?  They hurt performance under load.
+    // Why not ROCKSDB and KVSTORE_ROCKSDB_STATS, KVSTORE_PRIVATE_ROCKSDB_STATS?  They hurt
```
Comment (CjHare) [inline metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java]: Comment might be wrong, where is the `ROCKSDB` in the enum?

File: plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactoryInterface.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+package org.hyperledger.besu.plugin.services.storage.rocksdb;
+
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.storage.rocksdb.configuration.RocksDBConfiguration;
+
+import org.rocksdb.Statistics;
+import org.rocksdb.TransactionDB;
+
+public interface RocksDBMetricsFactoryInterface {
```
Comment (CjHare) [inline plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactoryInterface.java]: The `Interface` part of the name seem redundant on an `interface`

File: metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java (hunk 1 above)
Comment (rain-on) [reply metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java]: yeah ... it probably is :) nice.

File: plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactoryInterface.java (hunk 3 above)
Comment (ajsutton) [reply plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactoryInterface.java]: I'm not sure we need this interface at all. I may be missing something though.

**Comment by rain-on** [comment]
The interface is legacy ... thought I was going to need to hack the Plugin-api ... but avoided it :)

File: plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactory.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.Statistics;
+import org.rocksdb.TransactionDB;
+
+public class RocksDBMetricsFactory implements RocksDBMetricsFactoryInterface {
+
+  public static RocksDBMetricsFactory PUBLIC_ROCKS_DB_METRICS =
```
Comment (jframe) [inline plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactory.java]: nit: these factory constants can be final

File: metrics/rocksdb/src/main/java/org/hyperledger/besu/metrics/rocksdb/RocksDBStats.java [hunk 5]
```diff
@@ -170,18 +171,24 @@
 
   public static void registerRocksDBMetrics(
```
Comment (jframe) [inline metrics/rocksdb/src/main/java/org/hyperledger/besu/metrics/rocksdb/RocksDBStats.java]: Looks like this is no longer used

File: metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java (hunk 2 above)
Comment (rain-on) [reply metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java]: done.

File: metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java (hunk 1 above)
Comment (rain-on) [reply metrics/core/src/main/java/org/hyperledger/besu/metrics/BesuMetricCategory.java]: done.

File: plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactoryInterface.java (hunk 3 above)
Comment (rain-on) [reply plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/RocksDBMetricsFactoryInterface.java]: file entirely removed.

**Pull request merged**